### PR TITLE
v2 post documentation

### PR DESCRIPTION
**Pull request by samweinberg23**

### CODE REVIEW

File: docs/api.md [hunk 1]
```diff
@@ -32,7 +32,7 @@ as `0`, otherwise your order will be rejected.
 
 After this is done, the order is ready to be submitted to the API.
 
-__Order fields__
+__V1 Order fields__
```
Comment (BrendanChou) [inline docs/api.md]: mark as deprecated here

File: docs/api.md [hunk 2]
```diff
@@ -70,6 +70,50 @@ Example:
...
+|expiration|string|The time in unix seconds at which this order will be expired and can no longer be filled. Use `"0"` to specify that there is no expiration on the order.|
+|salt|string|A random number to make the orderHash unique.|
+|typedSignature|string|The signature of the order.|
+
+Example:
+```json
+{
+    "isBuy": "true",
+    "isDecreaseOnly": "false",
+    "baseMarket": "WETH",
```
Comment (BrendanChou) [inline docs/api.md]: 0

File: docs/api.md [hunk 3]
```diff
@@ -70,6 +70,50 @@ Example:
...
+    "quoteMarket": "DAI",
```
Comment (BrendanChou) [inline docs/api.md]: 3

File: docs/api.md [hunk 4]
```diff
@@ -70,6 +70,50 @@ Example:
...
+    "amount": "10000000000",
+    "limitPrice": "20.3",
+    "triggerPrice": "0",
+    "limitFee": ".15",
```
Comment (BrendanChou) [inline docs/api.md]: 0.0015

File: docs/api.md [hunk 5]
```diff
@@ -70,6 +70,50 @@ Example:
...
+    "makerAccountNumber": "111",
```
Comment (BrendanChou) [inline docs/api.md]: just put 0 for the example

File: docs/api.md [hunk 6]
```diff
@@ -70,6 +70,50 @@ Example:
...
+    "makerAccountOwner": "0x3E5e9111Ae8eB78Fe1CC3bb8915d5D461F3Ef9A9",
+    "expiration": "4294967295",
+    "salt": "100",
+    "typedSignature": "0xd9c006cf9066e89c2e75de72604751f63985f173ca3c69b195f1f5f445289a1f2229c0475949858522c821190c5f1ec387f31712bd21f6ac31e4510d5711c2681f00"
+  },
+};
+
+Note: The tick size is 0.01 for ETH-DAI transactions and 0.01e-12 for ETH-USDC and DAI-USDC transactions.
+The `limitPrice` must be divisible by the tick size.
+If `triggerPrice` is set, it must be divisible by the ticks size.
```
Comment (BrendanChou) [inline docs/api.md]: nit: ticks

File: docs/api.md [hunk 7]
```diff
@@ -70,6 +70,50 @@ Example:
...
+    "limitFee": ".15",
+    "makerAccountNumber": "111",
+    "makerAccountOwner": "0x3E5e9111Ae8eB78Fe1CC3bb8915d5D461F3Ef9A9",
+    "expiration": "4294967295",
+    "salt": "100",
+    "typedSignature": "0xd9c006cf9066e89c2e75de72604751f63985f173ca3c69b195f1f5f445289a1f2229c0475949858522c821190c5f1ec387f31712bd21f6ac31e4510d5711c2681f00"
+  },
+};
+
+Note: The tick size is 0.01 for ETH-DAI transactions and 0.01e-12 for ETH-USDC and DAI-USDC transactions.
```
Comment (BrendanChou) [inline docs/api.md]: 0.0001e-12 for DAI-USDC

Comment (BrendanChou) [reply docs/api.md]: Also explain the e-12 for this and the price

File: docs/api.md [hunk 8]
```diff
@@ -172,19 +218,20 @@ Example Request Body:
       	"fillOrKill": true,
       	"cancelAmountOnRevert": true,
       	"postOnly": false,
-      	"triggerPrice": "10100000000",
+       "triggerPrice": "0",
+  	"cancelId": "false",
```
Comment (BrendanChou) [inline docs/api.md]: nit: indentation for this whole section

File: docs/api.md [hunk 9]
```diff
@@ -172,19 +218,20 @@ Example Request Body:
...
 	"clientId": "foo",
 	"order": {
-		"makerMarket": "0",
-		"takerMarket": "1",
-		"makerAmount": "10000000000",
-		"takerAmount": "20000000000",
+    		"isBuy": "true",
+    		"isDecreaseOnly": "false",
+    		"baseMarket": "WETH",
+    		"quoteMarket": "DAI",
```
Comment (BrendanChou) [inline docs/api.md]: 3 and 0

File: docs/api.md [hunk 10]
```diff
@@ -172,19 +218,20 @@ Example Request Body:
...
+    		"amount": "10000000000",
+    		"limitPrice": "20.3",
+    		"triggerPrice": "0",
+    		"limitFee": ".15",
+    		"makerAccountNumber": "111",
```
Comment (BrendanChou) [inline docs/api.md]: fix these values to be same as above

File: docs/typescript.md [hunk 11]
```diff
@@ -253,6 +253,55 @@ const { order } = await solo.api.placeOrder({
...
+    side: ApiSide.BUY
+
+    //Market the trade is in
+    market: ApiMarketName.WETH_DAI
+
+    // denominated in base units. i.e. 1 ETH = 1e18
+    amount: new BigNumber('1e18'),
+
+    // denominated in base/quote.
+    price: '2.1',
```
Comment (BrendanChou) [inline docs/typescript.md]: make the price closer to actual price of ETH, also explain why you would want 'e-12' for USDC markets

File: docs/typescript.md [hunk 12]
```diff
@@ -253,6 +253,55 @@ const { order } = await solo.api.placeOrder({
...
+    makerAccountOwner: '0x52bc44d5378309ee2abf1539bf71de1b7d7be3b5',
+
+    // OPTIONAL: number of seconds until the order expires.
+    // 0 indicates no expiry. Defaults to 28 days
+    expiration: new BigNumber('1000'),
+
+    //OPTIONAL: Maximum fee you are willing to accept. Note, if limitFee is below calculated restriction and no exemption was given, the request will 400
+    //Makers will pay 0% fees. Takers with greater than or equal to .5Eth in the transaction will pay .15% of ETH-DAI and ETH-USDC transactions and .05% for DAI-USDC transactions.
+    //For transactions below .5Eth they will pay .50% fees.
+    limitFee: '.15'
```
Comment (BrendanChou) [inline docs/typescript.md]: 0.0015 is 0.15%

File: docs/typescript.md [hunk 13]
```diff
@@ -253,6 +253,55 @@ const { order } = await solo.api.placeOrder({
...
+  fillOrKill: false,
+
+  // OPTIONAL: defaults to false
+  postOnly: false,
+
+  // OPTIONAL: defaults to undefined
+  clientId: 'foo',
+
+  // OPTIONAL: order to be replaced
+  cancelId: '0x2c45cdcd3bce2dd0f2b40502e6bea7975f6daa642d12d28620deb18736619fa2',
```
Comment (BrendanChou) [inline docs/typescript.md]: explain this a bit more (that it allows you to replace an order)

File: docs/api.md (hunk 7 above)
Comment (samweinberg23) [reply docs/api.md]: Not sure what the best explanation would be - the decimal relationship of USDC vs. ETH and DAI?

File: docs/api.md [hunk 14]
```diff
@@ -70,6 +70,50 @@ Example:
...
+|makerAccountOwner|string|The Ethereum address of the Maker.|
+|expiration|string|The time in unix seconds at which this order will be expired and can no longer be filled. Use `"0"` to specify that there is no expiration on the order.|
+|salt|string|A random number to make the orderHash unique.|
+|typedSignature|string|The signature of the order.|
+
+Example:
+```json
+{
+    "isBuy": "true",
+    "isDecreaseOnly": "false",
```
Comment (BrendanChou) [inline docs/api.md]: nit: I believe these are not strings and are just `true` and `false`

File: docs/api.md [hunk 15]
```diff
@@ -70,6 +70,50 @@ Example:
...
+    "baseMarket": "0",
+    "quoteMarket": "3",
+    "amount": "10000000000",
+    "limitPrice": "20.3",
+    "triggerPrice": "0",
+    "limitFee": ".0015",
```
Comment (BrendanChou) [inline docs/api.md]: I think this would typically be stringified as `"0.0015"`

File: docs/api.md [hunk 16]
```diff
@@ -70,6 +70,50 @@ Example:
...
+|----------|---------|-----------|
+|isBuy|boolean|If the order is a buy order|
+|isDecreaseOnly|boolean|(Optional)If the Stop-Limit order is tied to an existing Isolated Position|
+|baseMarket|string|The Solo base [market](https://docs.dydx.exchange/#/overview?id=markets)|
+|quoteMarket|string|The Solo quote [market](https://docs.dydx.exchange/#/overview?id=markets)|
+|amount|string|The amount of token being offered in base units|
+|limitPrice|string| The worst base/quote price at which the transaction will be accepted|
+|triggerPrice|string|(Optional)The price at which the order will go to market.|
+|limitFee|string| Makers will pay 0% fees. Takers with greater than or equal to .5Eth in the transaction will pay .15% of ETH-DAI and ETH-USDC transactions and .05% for DAI-USDC transactions.
+For transactions below .5Eth they will pay .50% fees.
```
Comment (BrendanChou) [inline docs/api.md]: nit: for all percentages, put the 0 before the decimal place

File: docs/api.md [hunk 17]
```diff
@@ -70,6 +70,50 @@ Example:
...
+    "limitFee": ".0015",
+    "makerAccountNumber": "0",
+    "makerAccountOwner": "0x3E5e9111Ae8eB78Fe1CC3bb8915d5D461F3Ef9A9",
+    "expiration": "4294967295",
+    "salt": "100",
+    "typedSignature": "0xd9c006cf9066e89c2e75de72604751f63985f173ca3c69b195f1f5f445289a1f2229c0475949858522c821190c5f1ec387f31712bd21f6ac31e4510d5711c2681f00"
+  },
+};
+
+Note: The tick size is 0.01 for ETH-DAI, 0.01e-12 for ETH-USDC and 0.0001e-12 for DAI-USDC transactions. The negative twelfth power is because USDC has 8 decimal places of specificity whereas ETH and DAI have 18. This tick check prevents unreasonable values from being set
```
Comment (BrendanChou) [inline docs/api.md]: USDC has 6.

18 - 6 = 12

Comment (BrendanChou) [reply docs/api.md]: Can remove ` This tick check prevents unreasonable values from being set`

File: docs/api.md [hunk 18]
```diff
@@ -172,19 +218,20 @@ Example Request Body:
       	"fillOrKill": true,
       	"cancelAmountOnRevert": true,
       	"postOnly": false,
-      	"triggerPrice": "10100000000",
+      	"triggerPrice": "0",
+	"cancelId": "false",
```
Comment (BrendanChou) [inline docs/api.md]: `cancelId` would not appear in the request. Otherwise, put an actual hash

File: docs/api.md [hunk 19]
```diff
@@ -172,19 +218,20 @@ Example Request Body:
...
 	"clientId": "foo",
 	"order": {
-		"makerMarket": "0",
-		"takerMarket": "1",
-		"makerAmount": "10000000000",
-		"takerAmount": "20000000000",
+      		"isBuy": "true",
+      		"isDecreaseOnly": "false",
```
Comment (BrendanChou) [inline docs/api.md]: true/false  not strings

File: docs/api.md [hunk 20]
```diff
@@ -172,19 +218,20 @@ Example Request Body:
...
+      		"baseMarket": "0",
+      		"quoteMarket": "3",
+      		"amount": "10000000000",
+      		"limitPrice": "20.3",
+      		"triggerPrice": "0",
+      		"limitFee": ".0015",
```
Comment (BrendanChou) [inline docs/api.md]: nit: 0.0015

File: docs/typescript.md [hunk 21]
```diff
@@ -253,6 +253,55 @@ const { order } = await solo.api.placeOrder({
...
+    //Side your order is being placed on
+    side: ApiSide.BUY
+
+    //Market the trade is in
+    market: ApiMarketName.WETH_DAI
+
+    // denominated in base units. i.e. 1 ETH = 1e18
+    amount: new BigNumber('1e18'),
+
+    // denominated in base/quote. Since ETH and DAI have 18 decimals and are represented in e-18 while USDC has 8 decimals and is e-8, USDC amounts are represented in e-12
```
Comment (BrendanChou) [inline docs/typescript.md]: 6 instead of 8

Comment (BrendanChou) [reply docs/typescript.md]: nit: "USDC prices" instead of "USDC amounts"

File: docs/typescript.md [hunk 22]
```diff
@@ -253,6 +253,55 @@ const { order } = await solo.api.placeOrder({
...
+    makerAccountOwner: '0x52bc44d5378309ee2abf1539bf71de1b7d7be3b5',
+
+    // OPTIONAL: number of seconds until the order expires.
+    // 0 indicates no expiry. Defaults to 28 days
+    expiration: new BigNumber('1000'),
+
+    //OPTIONAL: Maximum fee you are willing to accept. Note, if limitFee is below calculated restriction and no exemption was given, the request will 400
+    //Makers will pay 0% fees. Takers with greater than or equal to .5Eth in the transaction will pay .15% of ETH-DAI and ETH-USDC transactions and .05% for DAI-USDC transactions.
+    //For transactions below .5Eth they will pay .50% fees.
+    limitFee: '.0015'
```
Comment (BrendanChou) [inline docs/typescript.md]: nit: 0.0015 for consistency

**Pull request merged**